corehq/apps/users/tests/test_models_role.py

```diff
@@ -0,0 +1,263 @@
+from django.test import TestCase
+from field_audit.models import AuditEvent
+from nose.tools import nottest
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models import Permissions
+
+from ..models_role import (
+    RoleAssignableBy,
+    RolePermission,
+    SQLPermission,
+    UserRole,
+)
+
+
+class CaseWithDomainAndUser(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        AuditEvent.objects.all().delete()
+        cls.domain_name = "pets"
+        cls.domain = Domain(name=cls.domain_name)
+        cls.domain.save()
+        cls.role = UserRole.create(
+            cls.domain_name,
+            "test-class-role",
+            permissions=Permissions.min(),  # init with no permissions
+        )
+        cls.permission = SQLPermission.objects.create(value="do_bupkis")
+        cls.class_audit_event_ids = {e.id for e in AuditEvent.objects.all()}
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain.delete()
+        super().tearDownClass()
+
+    @nottest
+    def get_test_audit_events(self, ignore_ids=[]):
+        return AuditEvent.objects.exclude(
+            id__in=self.class_audit_event_ids.union(ignore_ids),
+        )
+
+
+class TestRoleAssignableBy(CaseWithDomainAndUser):
+
+    def setUp(self):
+        self.role.set_assignable_by([self.role.id])
+        event, = self.get_test_audit_events()
+        self.assignment = RoleAssignableBy.objects.get(id=event.object_pk)
+
+    def test_audit_fields_for_assignableby_create(self):
+        setup_event, = self.get_test_audit_events()
+        self.assertTrue(setup_event.is_create)
+        self.assertEqual(
+            "corehq.apps.users.models_role.RoleAssignableBy",
+            setup_event.object_class_path,
+        )
+        self.assertEqual(
+            {
+                "role": {"new": self.role.id},
+                "assignable_by_role": {"new": self.role.id},
+            },
+            setup_event.delta,
+        )
+
+    def test_audit_fields_for_assignableby_delete(self):
+        remember = self.assignment.pk
+        setup_event, = self.get_test_audit_events()
+        self.assignment.delete()
```

Comment:
The same applies to permissions

Reply:
@calellowitz also mentioned the `QuerySet` write methods on dimagi/django-field-audit#1. It will be addressed [soon in another PR](https://github.com/dimagi/django-field-audit/pull/1#discussion_r919305129).